subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -404,8 +404,8 @@ private void createOrdinalRelationships(Node node) {
 
             if (taskClassifier.isDestroyer()) {
                 // Create (or get) a destroyer ordinal node that depends on the dependencies of this task node
-                Node ordinalNode = ordinalNodeAccess.getOrCreateDestroyableLocationNode(taskNode.getOrdinal());
-                taskNode.getHardSuccessors().forEach(ordinalNode::addDependencySuccessor);
+                OrdinalNode ordinalNode = (OrdinalNode) ordinalNodeAccess.getOrCreateDestroyableLocationNode(taskNode.getOrdinal());
```

Comment:
Is the downcast always safe?

Reply:
Yep.  Should be.  I suppose `OrdinalNodeAccess` should always return `OrdinalNode`s, though.